tools/tc39-test262/src/test_files.rs

```diff
@@ -0,0 +1,125 @@
+use std::path::Path;
+use rustc_hash::{FxHashMap, FxHashSet};
+use serde::Deserialize;
+
+use crate::{TestFlags, SpecEdition};
+use super::structs::*;
+
+/// All the harness include files.
+#[allow(missing_docs)]
+#[derive(Debug, Clone)]
+pub struct Harness {
+    pub assert: HarnessFile,
+    pub sta: HarnessFile,
+    pub doneprint_handle: HarnessFile,
+    pub includes: FxHashMap<Box<str>, HarnessFile>,
+}
+
+/// Represents harness file.
+#[allow(missing_docs)]
+#[derive(Debug, Clone)]
+pub struct HarnessFile {
+    pub content: Box<str>,
+    pub path: Box<Path>,
+}
+
+/// Represents a test.
+#[allow(dead_code)]
+#[allow(missing_docs)]
+#[derive(Debug, Clone)]
+pub struct Test {
+    pub name: Box<str>,
+    pub path: Box<Path>,
+    pub description: Box<str>,
+    pub esid: Option<Box<str>>,
+    pub edition: SpecEdition,
+    pub flags: TestFlags,
+    pub information: Box<str>,
+    pub expected_outcome: Outcome,
+    pub features: FxHashSet<Box<str>>,
+    pub includes: FxHashSet<Box<str>>,
+    pub locale: Locale,
+    pub ignored: bool,
+}
+
+impl Test {
+    /// Creates a new test.
+    pub fn new<N, C>(name: N, path: C, metadata: MetaData) -> color_eyre::Result<Self>
```

Comment:
We should not use `eyre` for error handling. We should have an `Error` enum that people can match on.